data-prepper-plugins/grok-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/grok/GrokProcessor.java

```diff
@@ -319,7 +319,12 @@ private boolean shouldBreakOnMatch(final Map<String, Object> captures) {
     }
 
     private void runWithTimeout(final Runnable runnable) throws TimeoutException, ExecutionException, InterruptedException {
-        Future<?> task = executorService.submit(runnable);
-        task.get(grokProcessorConfig.getTimeoutMillis(), TimeUnit.MILLISECONDS);
+        final Future<?> task = executorService.submit(runnable);
+        try {
+            task.get(grokProcessorConfig.getTimeoutMillis(), TimeUnit.MILLISECONDS);
+        } catch (final TimeoutException exception) {
+            task.cancel(true);
+            throw exception;
+        }
```

Comment:
Should we keep a count of number of tasks submitted to the executor and export it as a metric?